### DISCUSSION TITLE
And interest in pyserial for brython?

### DISCUSSION THREAD

**Discussion by benjie-git** [brython-dev/brython#2404] [General]
I'm interested in porting pyserial to brython, to allow using serial ports from brython code, maintaining compatibility with running the same python code in cpython with pyserial.  This would use the Web Serial API, which is currently only available in Chromium-based browsers.

Would anyone else be interested in this?  And what would be the best way to manage this package?  Should it be a separate repo, or eventually included into brython, if accepted?

Thanks for your feedback!

**Comment by jsbueno** [comment]
is ti even possible?
I mean - javascript code in  a browser should not have low level access to
computer I/O ports - so can't Brython.

**Reply by denis-migdal** [reply]
It is, but not supported in all browser.
https://developer.mozilla.org/en-US/docs/Web/API/Web_Serial_API#browser_compatibility

Browser can access your devices (webcam, microphone, keyboard, mouse, GPU, etc.). The browser will ask you for permission first.
